ee-feature-pack/common/src/main/resources/modules/system/layers/base/org/jberet/jberet-jpa-repository/main/module.xml

```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2013, Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags. See the copyright.txt file in the
+  ~ distribution for a full listing of individual contributors.
+  ~
+  ~ This is free software; you can redistribute it and/or modify it
+  ~ under the terms of the GNU Lesser General Public License as
+  ~ published by the Free Software Foundation; either version 2.1 of
+  ~ the License, or (at your option) any later version.
+  ~
+  ~ This software is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.9" name="org.jberet.jberet-jpa-repository">
+
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${org.jberet:jberet-jpa-repository}"/>
+    </resources>
+
+    <dependencies>
+        <module name="jakarta.batch.api"/>
+        <module name="jakarta.persistence.api"/>
+        <module name="org.hibernate"/>
```

Comment:
What is the requirement for Hibernate here?

Reply:
@jamezp for this :
https://github.com/wildfly/wildfly/pull/16085/files#diff-fc7f3594782e45651276961fa54e3391cd3a12375cb38b6b2227a6fd93799107R55

Reply:
It looks like the value is never set so it could be removed. Ideally we aren't tied to an implementation here. Plus the implementation could be provided in the deployment meaning we would not get the correct `EntityManagerFactoryBuilder`.

Reply:
@jamezp , you're right about the fact that https://github.com/wildfly/wildfly/pull/16085/files#diff-fc7f3594782e45651276961fa54e3391cd3a12375cb38b6b2227a6fd93799107R73 is not needed anymore. It is something which remained from the very first (and initially failed) attempts to obtain an EntityManagerFactory in an agnostic way. Then, @amoscatelli succeded by means of this: https://github.com/wildfly/wildfly/pull/16085/files#diff-fc7f3594782e45651276961fa54e3391cd3a12375cb38b6b2227a6fd93799107R139

As you can see, those lines of code rely on the constant `HBM2DDL_AUTO`, whose value is `"spring.jpa.hibernate.ddl-auto"`, which has to be imported like this: https://github.com/wildfly/wildfly/pull/16085/files#diff-fc7f3594782e45651276961fa54e3391cd3a12375cb38b6b2227a6fd93799107R139
That's why the dependency on Hibernate seems to be still necessary (the underlying thought is to avoid using "magic strings" in the code and rely only on pre-defined values, so a compilation error will rise when that value is not supported again)

Reply:
IMO there should be no hard requirements on Hibernate. This could potentially break things if users include Hibernate in their deployment or they want to use a different Jakarta Persistence implementation.

WildFly has been working heavily on trying to reduce implementation dependencies between modules. This helps tools like WildFly Glow reduce the size of a provisioned server.